lua/neogit/lib/git/push.lua

```diff
@@ -14,7 +14,7 @@ function M.push_interactive(remote, branch, args)
 end
 
 local function update_unmerged(state)
-  if not state.upstream.branch then
+  if not state.upstream.ref then
```

Comment:
Why the general change from`branch` to `ref`?

Is it due to a collision with something else called `branch` or just to be consistent with git's naming?

Reply:
Partly to be more consistent with git nomenclature, and partly because I'm trying to use more internal state for things instead of calling out to the CLI. Some views make like... 5 cli calls that are exactly the same, to get stuff like branch names, or upstream names. So `upstream.ref` can be the unparsed version, and `upstream.branch` can be just the name.

Reply:
Oh in that case, fantastic.

Perhaps a comment that clears up their relationship (if there isn't one already) would be a good idea.

Reply:
Yeah, I'll add documentation in the repository that lays out what everything is. Good call.